### PR TITLE
ltbi results postprocessing

### PR DESCRIPTION
**Pull request by yongqx**
Compile updates on James' scripts.

### CODE REVIEW

**Comment by collijk** [comment]
Thanks for moving this all into a script!  I wrote the results processing off some data I preprocessed into indexed data frames.  Does this work with the raw output.hdfs?  Otherwise, I think I have a script sitting in one of the output dirs I need to incorporate here.

**Comment by yongqx** [comment]
> Thanks for moving this all into a script! I wrote the results processing off some data I preprocessed into indexed data frames. Does this work with the raw output.hdfs? Otherwise, I think I have a script sitting in one of the output dirs I need to incorporate here.

I also pushed the scripts for preprocessing the raw output.hdf and it generates exactly same country specific indexed dataframe as we had before. I will test the scripts and see whether it works for new results later!

File: src/vivarium_csu_ltbi/tools/ltbi_results_postprocessing.py [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+    
+    index_columns = ['outcome', 'location', 'year', 'age', 'sex', 'risk_group', 'scenario', 'treatment_group']
+    raw_scaled_summary = pivot_and_summarize(raw_scaled, index_columns)
+    delta_scaled_summary = pivot_and_summarize(delta_scaled, index_columns, prefix='averted_')
+    scaled_summary = pd.concat([raw_scaled_summary, delta_scaled_summary], axis=1)
+    
+    return pd.concat([counts_summary, scaled_summary])
+
+
+def make_deaths_table(mdata: MeasureData):
```
Comment (yongqx) [inline src/vivarium_csu_ltbi/tools/ltbi_results_postprocessing.py:273]: to-do: add counts_summary to the death table.

File: src/vivarium_csu_ltbi/tools/ltbi_results_postprocessing.py [hunk 2]
```diff
@@ -0,0 +1,364 @@
+import numpy as np
+import pandas as pd
+import warnings
+warnings.filterwarnings('ignore')
+
+from typing import NamedTuple
+
+output_dir = '/home/j/Project/simulation_science/latent_tuberculosis_infection/result/'
+
+def make_raw_aggregates(data):
```
Comment (yongqx) [inline src/vivarium_csu_ltbi/tools/ltbi_results_postprocessing.py:10]: to-do: relabel the age group.

Comment (yongqx) [reply src/vivarium_csu_ltbi/tools/ltbi_results_postprocessing.py:10]: change 0_to_5 to 0_to_4, 5_to_15 to 5_to_14, 15_to_60 to 15_to_59.

File: src/vivarium_csu_ltbi/tools/ltbi_results_postprocessing.py [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+    raw_scaled = scale_data(pt, pt, scale_join_columns, ['treatment_group'])
+    delta_scaled = scale_data(pt, delta, scale_join_columns, ['treatment_group'])
+    
+    index_columns = ['outcome', 'location', 'year', 'age', 'sex', 'risk_group', 'scenario', 'treatment_group']
+    raw_summary = pivot_and_summarize(raw_scaled, index_columns)
+    delta_summary = pivot_and_summarize(delta_scaled, index_columns, prefix='averted_')
+    
+    return pd.concat([raw_summary, delta_summary], axis=1)
+
+def make_tb_table(mdata: MeasureData):
```
Comment (yongqx) [inline src/vivarium_csu_ltbi/tools/ltbi_results_postprocessing.py:240]: use `all_person_time` as denominator to calculate `actb_incidence_rate`

File: src/vivarium_csu_ltbi/tools/ltbi_results_postprocessing.py [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+        for location in ['ethiopia', 'india', 'peru', 'philippines', 'south_africa']:
+            df = pd.read_hdf(output_dir + f'sim_raw_hdf/{location}_{age_end}_indexed.hdf').reset_index()
+            df = make_raw_aggregates(df)
+            measure_data = split_measures(df, location)
+            loc_data = []
+            for f in [make_coverage_table, make_tb_table, make_deaths_table, make_dalys_table, make_person_time_table]:
+                loc_data.append(f(measure_data))
+            output.append(pd.concat(loc_data))
+        pd.concat(output).to_csv(output_dir + f'ltbi_final_results_{age_end}.csv')
+        pd.concat(output).to_hdf(output_dir + f'ltbi_final_results_{age_end}.hdf', key='data')
```
Comment (yongqx) [inline src/vivarium_csu_ltbi/tools/ltbi_results_postprocessing.py:364]: add `mode='w'` when saving hdf files.

**Pull request merged**